internal/otel/config.go

```diff
@@ -18,13 +18,13 @@ func newProvider(cfg CollectorCfg) (otelcol.ConfigProvider, error) {
 	if cfg.ResourceID != "" {
 		uris = append(uris, fmt.Sprintf("hcp:%s", cfg.ResourceID))
 	}
+
 	resolver := confmap.ResolverSettings{
 		URIs: uris,
 		Providers: makeMapProvidersMap(
-			external.NewProvider(cfg.ForwarderEndpoint),
-			hcp.NewProvider(cfg.ForwarderEndpoint, cfg.Client, cfg.ClientID, cfg.ClientSecret),
+			external.NewProvider(cfg.ExporterConfig),
+			hcp.NewProvider(cfg.ExporterConfig, cfg.Client, cfg.ClientID, cfg.ClientSecret),
```

Comment:
jw, looks like we currently allow running the collector without cloud enabled / having HCP variables based not these `agent/service.go`lines:

https://github.com/hashicorp/consul-telemetry-collector/pull/88/files#diff-a24b268fee3eff44953ecb95bdfa16489b4bcc8e4e7e5111a71dd4438aeb196bR40-R53

Is it ok to still create a provider if this hasn't been enabled? I'm assuming that `Retrieve()` function below would fail. Did we ever do a manual test for this... I might totally be missing OTEL context btw  🤔

Reply:
Actually, same with the external provider, do we need to initialize it if we don't have any external exports?

Reply:
Ah ok, is this where we do the checks?
https://github.com/hashicorp/consul-telemetry-collector/pull/88/files#diff-eff9326a222a55b0adc5abad7a90af0bbdd6db52b82d9ab5a3ca7bfd3ea1cd4dR53-R57

Reply:
Calling the `Retrieve` function is handled by passing the appropriate `scheme` into the `confmap.Resolver` here: https://github.com/hashicorp/consul-telemetry-collector/blob/8b03bae3d15cc7e4216295eecb7dbed5b1a2eaf6/internal/otel/config.go#L17